functions/go-mail/handler.go

```diff
@@ -0,0 +1,125 @@
+package gomail
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/go-playground/validator/v10"
+	tem "github.com/scaleway/scaleway-sdk-go/api/tem/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+// Body to send to the function (via curl for example or Messaging & Queuing).
+
+//	{
+//		"to": "mail@mail.mail",
+//		"subject": "from console test",
+//		"message": "very very long msg"
+//	}
+
+// region used to call the API.
+const region = scw.RegionFrPar
+
+// Data is a struct to handle the body of the HTTP call. Fields in Data must be completed
+// to send and email.
+type Data struct {
+	Subject string `json:"subject" 	validate:"required"`
+	Message string `json:"message" 	validate:"required"`
+	To      string `json:"to" 		validate:"required,email"`
+}
+
+// Handler is the entrypoint of the function.
+func Handler(respWriter http.ResponseWriter, req *http.Request) {
+	// Only allow POST verbs
+	if req.Method != http.MethodPost {
+		respWriter.WriteHeader(http.StatusMethodNotAllowed)
+
+		return
+	}
+
+	var body Data
+
+	if err := json.NewDecoder(req.Body).Decode(&body); err != nil {
+		respWriter.WriteHeader(http.StatusBadRequest)
+
+		return
+	}
+
+	// Validate the Data structure, return a 400 error if not valid.
+
+	if err := validator.New().Struct(body); err != nil {
+		respWriter.WriteHeader(http.StatusBadRequest)
+		_, _ = respWriter.Write([]byte(err.Error()))
+
+		return
+	}
+
+	if err := sendMail(body.Subject, body.Message, body.To, "CHANGE_ME", false); err != nil {
+		panic(err)
+	}
+}
+
+// sendMail function is used to send a mail to "mailTo" using "from" email.
+// If checkMailStatus is at true, the function will take more tame in order to run some calls
+// to the API in order to get mail status over time.
+func sendMail(subject, content, mailTo, from string, checkMailStatus bool) error {
+	// Create a Scaleway client
+	client, err := scw.NewClient(
+		// Get your organization ID at https://console.scaleway.com/organization/settings
+		scw.WithDefaultOrganizationID(os.Getenv("SCW_DEFAULT_ORGANIZATION_ID")),
```

Comment:
Can you add a note in the README to explain that these variables should be defined before deploying the function or else it won't work

Reply:
Yes I added it to serverless.yml and readme 👌